File: app/src/main/java/com/wkz/viewpager/transformer/RotateUpTransformer.java
package com.wkz.viewpager.transformer;

import android.view.View;

import com.nineoldandroids.view.ViewHelper;

/**
 * @desc 右转一定角度的转换效果
 * @date 2017-06-30
 */
public class RotateUpTransformer extends AbstractBaseTransformer {

    private static final float ROT_MOD = -15F;

    @Override
    protected void onTransform(View page, float position) {
        float pageWidth = page.getWidth();
        float rotation = ROT_MOD * position;

        ViewHelper.setPivotX(page, pageWidth * 0.5F);
        ViewHelper.setPivotY(page, 0F);
        ViewHelper.setTranslationX(page, 0F);
        ViewHelper.setRotation(page, rotation);
    }

    @Override
    protected boolean isPagingEnabled() {
        return true;
    }
}

File: view-pager-transformer/src/main/java/com/fphoenixcorneae/transformer/RotateUpTransformer.kt
package com.fphoenixcorneae.transformer

import android.view.View

/**
 * @desc 右转一定角度的转换效果
 * @date 2017-06-30
 */
class RotateUpTransformer : AbstractBaseTransformer() {

    override val isPagingEnabled: Boolean = true

    override fun onTransform(page: View, position: Float) {
        val pageWidth = page.width.toFloat()
        page.apply {
            pivotX = pageWidth * 0.5f
            pivotY = 0f
            translationX = 0f
            rotation = ROT_MOD * position
        }
    }

    companion object {
        private const val ROT_MOD = -15f
    }
}